db/crud_test.go

```diff
@@ -956,3 +956,211 @@ func BenchmarkHandleRevDelta(b *testing.B) {
+	assert.NotEmpty(t, doc2, "Document shouldn't be empty")
+	assert.Equal(t, "2-a", rev2, "Provided input revision ID should be returned")
+
+	// Get the 1x revision from second doc.
+	bodyBytes, removed, err := db.get1xRevFromDoc(doc2, rev2, true)
+	assert.NotEmpty(t, bodyBytes, "Document body bytes should be returned")
+	assert.False(t, removed, "This shouldn't be a removed document")
+	var response = Body{}
+	assert.NoError(t, response.Unmarshal(bodyBytes))
+	assert.Equal(t, "camera", response[BodyId])
+	assert.Equal(t, "2-a", response[BodyRev])
+	assert.Equal(t, "1483C002SKU", response["model"])
+	assert.Equal(t, "Canon EOS 5D Mark IV", response["name"])
+	assert.Equal(t, "$2,499.99", response["price"])
+	assert.Equal(t, "5578523", response["sku"])
+	assert.Equal(t, "2a", response["version"])
+
+	// Get 1x revision from doc with unknown revision id; it simulates the error scenario.
+	// A 404 missing error should be thrown when trying get the body bytes of the document
+	// which doesn't exists in the database. The revision "3-a" doesn't exists in database.
+	bodyBytes, removed, err = db.get1xRevFromDoc(doc2, "3-a", true)
+	assert.Error(t, err, "It should throw 404 missing error")
+	assert.Contains(t, err.Error(), "404 missing")
+	assert.Empty(t, bodyBytes, "Provided revision doesn't exists")
+	assert.False(t, removed, "This shouldn't be a removed revision")
+	assert.Error(t, response.Unmarshal(bodyBytes), "Unexpected empty JSON input to body.Unmarshal")
+
+	// Create the third revision of the document
+	payload = `{"model":"1483C002SKU","sku":"5578523","name":"Canon EOS 5D Mark IV","price":"$2,499.99","version":"3a"}`
+	doc3, rev3, err := db.PutExistingRevWithBody("camera", unjson(payload), []string{"3-a", "2-a"}, false)
+	assert.NoError(t, err, "Couldn't create document")
+	assert.NotEmpty(t, doc3, "Document shouldn't be empty")
+	assert.Equal(t, "3-a", rev3, "Provided input revision ID should be returned")
+
+	// Get the 1x revision from third doc.
+	bodyBytes, removed, err = db.get1xRevFromDoc(doc3, rev3, true)
+	assert.NotEmpty(t, bodyBytes, "Document body bytes should be returned")
+	assert.False(t, removed, "This shouldn't be a removed document")
+	assert.NoError(t, response.Unmarshal(bodyBytes))
+	assert.Equal(t, "camera", response[BodyId])
+	assert.Equal(t, "3-a", response[BodyRev])
+	assert.Equal(t, "1483C002SKU", response["model"])
+	assert.Equal(t, "Canon EOS 5D Mark IV", response["name"])
+	assert.Equal(t, "$2,499.99", response["price"])
+	assert.Equal(t, "5578523", response["sku"])
+	assert.Equal(t, "3a", response["version"])
+
+	// Create the fourth revision by deleting the third revision
+	rev4, err := db.DeleteDoc("camera", rev3)
+	assert.NoError(t, err, "Document should be deleted")
+	assert.NotEmpty(t, rev4, "Document revision shouldn't be empty")
+
+	// Get the document body bytes with the deleted revision, list revisions enabled.
+	bodyBytes, removed, err = db.get1xRevFromDoc(doc3, rev3, true)
```

Comment:
rev3 shouldn't be a tombstone, should it?  I'd expect rev4 to be the tombstone.

Reply:
Yeah, you are right, changes are committed.